pkg/codegen/schema/schema.go

```diff
@@ -1129,33 +1182,69 @@ func (t *types) bindObjectType(token string, spec ObjectTypeSpec) (*ObjectType,
 	return obj, nil
 }
 
-func bindTypes(pkg *Package, objects map[string]ObjectTypeSpec) (*types, error) {
+func (t *types) bindEnumTypeDetails(enum *EnumType, token string, spec ComplexTypeSpec) error {
+	enum.Token = token
+	enum.Elements = t.bindEnumValues(spec.Enum)
+	enum.ElementType = spec.Type
+	enum.Comment = spec.Description
+
+	return nil
+}
+
+func (t *types) bindEnumValues(values []*EnumValueSpec) (enums []*Enum) {
+	for _, spec := range values {
+		enum := &Enum{
+			Value:       spec.Value,
```

Comment:
It would be good to check that the type of `Value` is assignable to the element type of the enum. You can do that in a follow-up change, though.